test/TestTransmuteDPOS.spec.js

```diff
@@ -347,6 +344,12 @@ contract('TransmuteDPOS', accounts => {
       assert.equal(newAmount, (await tdpos.getProvider(accounts[0]))[0].toNumber()); // [0] is totalAmountBonded
     });
 
+    it('should resign the Provider if a Provider calls the function', async () => {
+      assert.equal(true, await tdpos.containsProvider(accounts[1]));
+      await tdpos.unbond({from: accounts[1]});
+      assert.equal(false, await tdpos.containsProvider(accounts[1]));
+    });
+
     it('should remove the Delegator from the mapping', async () => {
       let delegator = await tdpos.delegators.call(accounts[4]);
```

Comment:
It would be nice to add some variable names for accountAddresses that are more logic, such as the role an address is taking. accounts[5] is only really descriptive of a unique address, providerAccount or delegatorAccount, might be more descriptive, could make tests easier to read. these could be defined once at the top of the test, and could vary from scope block to scope block.